hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -428,6 +431,19 @@ public void bootstrap(Option<Map<String, String>> extraMetadata) {
    */
   public abstract O upsertPreppedRecords(I preppedRecords, final String instantTime);
 
+  /**
+   * Upserts the given prepared records into the Hoodie table, at the supplied instantTime.
+   * <p>
+   * This implementation requires that the input records are already tagged, and de-duped if needed.
+   *
+   * @param preppedRecords Prepared HoodieRecords to upsert
+   * @param instantTime Instant time of the commit
+   * @return Collection of WriteStatus to inspect errors and counts
+   */
+  public O upsertPreppedRecords(I preppedRecords, final String instantTime, Option<List<Pair<String, String>>> partitionFileIdPairsOpt) {
```

Comment:
Is there a way to avoid adding this to the base client now since it is specific to spark mdt writer?

Reply:
yes.